apps/server/src/service/dexLiquidityPoolService.ts

```diff
@@ -36,6 +36,29 @@ export class DexLiquidityPoolService {
     this.txBuilderServiceFactory = new txBuilder.TxBuilderServiceFactory();
   }
 
+  async poolInfoWithStatus(tokenAHash: string, tokenBHash: string): Promise<PoolInfo> {
+    const poolInfos = await this.getLiquidityPools();
+    const poolInfo = poolInfos.find((x) => {
+      const key1 = `${x.tokenA.typeHash}:${x.tokenB.typeHash}`;
```

Comment:
A refactoring of this code might make it easier to read

```ts
import { differenceWith, equals } from 'lodash';

poolInfos.find(
  (x) => differenceWith([tokenAHash, tokenBHash], [x.tokenA.typeHash, x.tokenB.typeHash], equals).length === 0,
);

```

Reply:
> A refactoring of this code might make it easier to read
> 
> ```ts
> import { differenceWith, equals } from 'lodash';
> 
> poolInfos.find(
>   (x) => differenceWith([tokenAHash, tokenBHash], [x.tokenA.typeHash, x.tokenB.typeHash], equals).length === 0,
> );
> ```

Currently it is a functional implementation, the subsequent matching algorithm needs to be modified